controller/should_namespace_sync.go

```diff
@@ -0,0 +1,41 @@
+package controller
+
+import (
+	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"reflect"
+)
+
+func shouldNamespaceSync(appLabelKey string, trackingMethod v1alpha1.TrackingMethod, appName string, syncPolicy *v1alpha1.SyncPolicy) func(un *unstructured.Unstructured) bool {
+	return func(un *unstructured.Unstructured) bool {
+		isNewNamespace := un != nil && un.GetUID() == "" && un.GetResourceVersion() == ""
+
+		/*
+			if isNewNamespace {
+				resourceTracking := argo.NewResourceTracking()
+				// TODO: do we need to handle this error?
```

Comment:
Yes.. We should handle the error returning in the internal function: `func(un *unstructured.Unstructured) (bool, error)`